apps/agent/README.md

```diff
@@ -0,0 +1,254 @@
+ - `balance(address _token)`
+
+More importantly, this property means that Agent is capable of using the DAO's funds to interact seemlessly with any contracts that require the sender to possess Ether or ERC-20 tokens (e.g. Exchanging tokens on [Uniswap](https://uniswap.io), participating in a [PoolTogether](https://www.pooltogether.com) lottery).
+
+### 1.4. Security
+
+Since the Agent app can manage tokens and Ether, it is crucial to always keep security in mind. In order to restrict unauthorized transfers and other abusive behaviors, a variant of the `execute` function is available that will prevent any attempt to transfer tokens included in a "protected tokens" list:
+
+- `safeExecute(address _target, bytes _data)` - Protected by the `SAFE_EXECUTE_ROLE` permission.
+
+Before executing the requested call, the function fetches in memory every balances from the protected tokens and compares them with the new balances after the call is executed. If a balance has changed, it reverts the call.
+
+The following functions can be used to add or remove a token address to the protected tokens list:
+
+- `addProtectedToken(address _token)` - Protected by the `ADD_PROTECTED_TOKEN_ROLE` permission.
+- `removeProtectedToken(address _token)` - Protected by the `REMOVE_PROTECTED_TOKEN_ROLE` permission.
+
+### 1.5. Forwarder
+
+
+## 2. Entry points
+
+### 2.1. Initialize
+ - **Actor:** External entity, ultimately calling the function through the `Kernel`
+ - **Inputs:** None
+ - **Pre-flight checks:**
+   - Ensure the app has not already been initialized
+ - **State transitions:**
+   - Initialize the app
+
+### 2.2. Execute
+ - **Actor:** External entity
+ - **Inputs:** 
+   - **Target:** Address where the action is being executed
+   - **ETH value:** Amount of ETH (in Wei) sent with the action
+   - **Data:** Calldata for the action
+ - **Pre-flight checks:**
+   - Ensure the actor has the `EXECUTE_ROLE` permission with the following ACL parameters:
+     - Target address
+     - ETH value
+     - Data signature
+ - **State transitions:**
+   - Decrease the contract's ETH amount by `ETH value` passed as input
+   - May increase or decrease the amount of tokens stored in the contract
+
+### 2.3. Safe execute
+ - **Actor:** External entity
+ - **Inputs:** 
+   - **Target:** Address where the action is being executed
+   - **Data:** Calldata for the action
+ - **Pre-flight checks:**
+   - Ensure the actor has the `SAFE_EXECUTE_ROLE` permission with the following ACL parameters:
+     - Target address
+     - Data signature
+ - **State transitions:** None
+
+### 2.4. Add protected token
+ - **Actor:** External entity
+ - **Inputs:** 
+   - **Token:** Address of the protected token
+ - **Pre-flight checks:**
+   - Ensure the actor has the `ADD_PROTECTED_TOKEN_ROLE` permission with the following ACL parameters:
+     - Token address
+   - Ensure that the current number of protected tokens is inferior than the maximum allowed (hardcoded as a constant)
```

Comment:
```suggestion
   - Ensure that the current number of protected tokens is less than the maximum allowed (hardcoded as a constant)
```

Maybe we should replace the "the maximum allowed" with the actual hardcoded number.

Reply:
Oops, classic french to english translation mistake 😆